packages/components/src/Spinner/src/Spinner.module.css

```diff
@@ -0,0 +1,83 @@
+@keyframes spin {
+    0% {
+        transform: rotate(0deg);
+    }
+
+    100% {
+        transform: rotate(360deg);
+    }
+}
+
+.hop-spinner {
+    --hop-spinner-component-border-radius: var(--hop-shape-circle);
+    --hop-spinner-component-sm-wheel-size: 1rem;
+    --hop-spinner-component-md-wheel-size: 1.25rem;
+    --hop-spinner-component-lg-wheel-size: 1.5rem;
+
+    /* Track */
+    --hop-spinner-component-track-width: 0.2rem;
+    --hop-spinner-component-track-opacity: 0.2;
+    --hop-spinner-component-track-border: var(--hop-spinner-component-track-width) solid currentcolor;
+
+    /* Fill */
+    --hop-spinner-component-fill-animation: spin .8s infinite linear;
+    --hop-spinner-component-fill-animation-reduced-motion-duration: 2s;
+    --hop-spinner-component-fill-border: var(--hop-spinner-component-track-width) solid transparent;
+    --hop-spinner-component-fill-border-inline-start: var(--hop-spinner-component-track-width) solid currentcolor;
+
+    /* Label */
+    --hop-spinner-component-gap: var(--hop-space-inline-sm);
+
+    box-sizing: border-box;
+    display: grid;
+    column-gap: var(--hop-spinner-component-gap);
+    grid-template-areas: "spinner label";
+    grid-template-columns: min-content auto;
+    align-items: center;
+}
+
+.hop-spinner--sm {
+    --width: var(--hop-spinner-component-sm-wheel-size);
+    --height: var(--hop-spinner-component-sm-wheel-size);
+}
+
+.hop-spinner--md {
+    --width: var(--hop-spinner-component-md-wheel-size);
+    --height: var(--hop-spinner-component-md-wheel-size);
+}
+
+.hop-spinner--lg {
+    --width: var(--hop-spinner-component-lg-wheel-size);
+    --height: var(--hop-spinner-component-lg-wheel-size);
+}
+
+.hop-spinner__track {
+    box-sizing: border-box;
+    inline-size: var(--width);
```

Comment:
where do the --width and -height variables come form and why are they so generic?

Reply:
They are designed just above. This is something francis showed me in order to better separate my css classes. I'll tag him to make sure it makes sense with him and to see if he wants to add something @fraincs